src/backend/web/routes/query.js

```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+
+const { logger } = require('../../utils/logger');
+const { search } = require('../../utils/elastic');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+  // Github rule, query is more than 256 chars return a validation failed message
+  if (req.query.q.length > 256) {
+    res.send('Validation failed, query length is greater than 256 characters.');
+    logger.info('Validation failed, query length is greater than 256 characters.');
+  }
+  try {
+    res.send(await search(req.query.q));
+  } catch (error) {
+    logger.error({ error }, 'Something went wrong with search indexing');
```

Comment:
Should we add an error response back to front-end so that it can display error? This seems like bad UX to not have some form of error feedback.

Reply:
Added a message in the form `res.send('There was an error while executing your query.');`